eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/EventsourcedProcessor.scala

```diff
@@ -113,7 +120,9 @@ trait EventsourcedProcessor extends EventsourcedWriter[Long, Long] with ActorLog
     case payload if processEvent.isDefinedAt(payload) =>
       val currentProcessedEvents = processEvent(payload)
       if (lastSequenceNr > processingProgress)
-        processedEvents = processedEvents :+ currentProcessedEvents.map(createEvent(_, lastHandledEvent.customDestinationAggregateIds))
+        processedEvents = processedEvents :+ currentProcessedEvents.map { e =>
+          postProcessDurableEvent(createEvent(e, lastHandledEvent.customDestinationAggregateIds))
```

Comment:
Calling `postProcessDurableEvent` causes the creation of a new function object for each processed event. Function object creation should be done once and only function application should be repeated. Although we don't have a behavior stack here, this is similar to what an `EventsourcedView` does for `onCommand`, `onEvent` and `onSnapshot` handlers (see https://github.com/RBMHTechnology/eventuate/blob/master/eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/EventsourcedView.scala#L113). Although not touched by your PR, the same problem exists for `processEvent` and should be changed accordingly. Such changes are a prerequisite for adding support for event processing [behavior changes](http://rbmhtechnology.github.io/eventuate/reference/event-sourcing.html#behavior-changes) later, if needed.